htdocs/comm/mailing/class/advtargetemailing.class.php

```diff
@@ -640,6 +640,10 @@ public function query_thirdparty($arrayquery)
 						if ($arrayquery['options_'.$key]!=''){
 							$sqlwhere[]= " (te.".$key." = ".$arrayquery['options_'.$key].")";
 						}
+					} elseif ($extrafields->attributes[$elementtype]['type'][$key] == 'link') {
+						if ($arrayquery['options_'.$key] > 0){
+                                                        $sqlwhere[]= " (te.".$key." = ".$arrayquery['options_'.$key].")";
```

Comment:
What is type of $arrayquery['options_'.$key]
If int add the cast (int) to avoid sql injection
If string, add ' ' and $this->db->escape()